iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/fileio/FileIOConfig.java

```diff
@@ -62,4 +64,23 @@ FileIO provideLocalFileIO() {
       return null;
     }
   }
+
+  /**
+   * Provides the ADLSFileIO bean for ADLS storage type
+   *
+   * @return ADLSFileIO bean for ADLS storage type, or null if ADLS storage type is not configured
+   */
+  @Bean
+  ADLSFileIO provideADLSFileIO() {
+    try {
+      // get the already-created FileIO from the ADLS Manager
+      ADLSFileIO fileIO =
+          ((AdlsStorageClient) storageManager.getStorage(StorageType.ADLS).getClient()).getFileIO();
```

Comment:
Can we create the ADLSFileIO object here, like we do S3FileIO?
https://github.com/linkedin/openhouse/blob/36aba11a0d4ba73204313a2735650f1f580e3afb/iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/fileio/FileIOConfig.java#L78

Reply:
Unfortunately, ADLSFileIO doesnt have the same constructor that allows us to pass in a created client. Instead, we have to initialize it with the inputted properties. [https://github.com/kmcclenn/openhouse/blob/ed0d2293904b995178c2614679b7c6aae3eb0d33/cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/adls/AdlsStorageClient.java#L65](https://github.com/kmcclenn/openhouse/blob/ed0d2293904b995178c2614679b7c6aae3eb0d33/cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/adls/AdlsStorageClient.java#L65)

Reply:
lets do following instead:
```
ADLSFileIO fileIO = (new ADLSFileIO()).initialize(AdlsStorage.getProperties())
```